conans/model/conan_file.py

```diff
@@ -318,6 +319,14 @@ def package_folder(self):
         """
         return self.folders.base_package
 
+    @property
+    def immutable_package_folder(self):
+        return self.folders.immutable_package_folder
+
+    @property
+    def install_folder(self):
+        return self.folders.install_folder
```

Comment:
This would only be useful inside the install() method to have an unambigous way to refer to the destionation folder of  the installed folder.

Are you proposing using immutable_package as the src, and package as the dst inside the install() method?

Reply:
Yes, using always the same naming as the consumer. If the consumer will see the folder as ``package_folder``, then the ``install()`` method should copy things from ``immutable_folder`` to ``package_folder``, to reduce the cognitive overload, of copying things to ``install_folder``, but then the consumer will not use the ``install_folder``, but the ``package_folder``.